sycl/test-e2e/Basic/nested_queue_submit.cpp

```diff
@@ -19,16 +26,44 @@ void nestedSubmit() {
   }
 }
 
-int main() {
+void nestedSubmitMemset(sycl::queue &q) {
+  uint32_t n = 1024;
+  int *data = sycl::malloc_device<int>(n, q);
+  try {
+    q.submit([&](sycl::handler &h) { q.memset(data, 0, n * sizeof(int)); });
+  } catch (...) {
+    sycl::free(data, q);
+    throw;
+  }
+  sycl::free(data, q);
+}
+
+void doValidCall(sycl::queue &q) {
+  q.submit([&](sycl::handler &h) {});
+}
+
+template <typename CommandSubmitterT>
+void test(sycl::queue &Queue, CommandSubmitterT QueueSubmit) {
+  bool ExceptionHappened = false;
   try {
-    nestedSubmit();
+    QueueSubmit(Queue);
   } catch (const sycl::exception &e) {
-    assert(e.code() == sycl::errc::invalid && "Invalid error code");
-    assert(std::string(e.what()) ==
-               "Calls to sycl::queue::submit cannot be nested. Command group "
-               "function objects should use the sycl::handler API instead." &&
-           "Invalid e.what() string");
+    checkExceptionFields(e);
+    ExceptionHappened = true;
   }
-  std::cout << "test passed" << std::endl;
+  assert(ExceptionHappened);
+  // Checks that queue is in a valid state and we could submit new commands.
+  doValidCall(Queue);
```

Comment:
I'm not very sure how does `doValidCall` validates the state of the queue. Adding an explanatory comment here will help.

Reply:
Also, can we rename `doValidCall` -> `checkQueueState` (Or something similar)?

Reply:
https://github.com/intel/llvm/pull/13659/commits/ec7c6b013e6e822a7d9319d2cc13b48862461370
https://github.com/intel/llvm/pull/13659/commits/783910da7743324d53fedd189d92d6e7262f56cd
I added comment and removed method call since now it consists of single line.